src/libraries/Common/src/System/Security/Cryptography/CompositeMLDsa.cs

```diff
@@ -0,0 +1,1800 @@
+
+            AsnWriter writer = ExportEncryptedPkcs8PrivateKeyCore(passwordBytes, pbeParameters);
+
+            // Skip clear since the data is already encrypted.
+            return EncodeAsnWriterToPem(PemLabels.EncryptedPkcs8PrivateKey, writer, clear: false);
+        }
+
+        /// <inheritdoc cref="ExportEncryptedPkcs8PrivateKey(ReadOnlySpan{char}, PbeParameters)"/>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="password"/> or <paramref name="pbeParameters"/> is <see langword="null"/>.
+        /// </exception>
+        public byte[] ExportEncryptedPkcs8PrivateKey(string password, PbeParameters pbeParameters)
+        {
+            ArgumentNullException.ThrowIfNull(password);
+
+            return ExportEncryptedPkcs8PrivateKey(password.AsSpan(), pbeParameters);
+        }
+
+        /// <summary>
+        ///   Exports the current key in the PKCS#8 EncryptedPrivateKeyInfo format with a byte-based password.
+        /// </summary>
+        /// <param name="passwordBytes">
+        ///   The bytes to use as a password when encrypting the key material.
+        /// </param>
+        /// <param name="pbeParameters">
+        ///   The password-based encryption (PBE) parameters to use when encrypting the key material.
+        /// </param>
+        /// <returns>
+        ///   A byte array containing the PKCS#8 EncryptedPrivateKeyInfo representation of the this key.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="pbeParameters"/> is <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="ObjectDisposedException">
+        ///   This instance has been disposed.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <para><paramref name="pbeParameters"/> specifies a KDF that requires a char-based password.</para>
+        ///   <para>-or-</para>
+        ///   <para><paramref name="pbeParameters"/> does not represent a valid password-based encryption algorithm.</para>
+        ///   <para>-or-</para>
+        ///   <para>This instance only represents a public key.</para>
+        ///   <para>-or-</para>
+        ///   <para>The private key is not exportable.</para>
+        ///   <para>-or-</para>
+        ///   <para>An error occurred while exporting the key.</para>
+        /// </exception>
+        public byte[] ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte> passwordBytes, PbeParameters pbeParameters)
+        {
+            ArgumentNullException.ThrowIfNull(pbeParameters);
+            PasswordBasedEncryption.ValidatePbeParameters(pbeParameters, ReadOnlySpan<char>.Empty, passwordBytes);
+            ThrowIfDisposed();
+
+            AsnWriter writer = ExportEncryptedPkcs8PrivateKeyCore(passwordBytes, pbeParameters);
+
+            try
+            {
+                return writer.Encode();
+            }
+            finally
+            {
+                writer.Reset();
```

Comment:
why is the reset needed? (here and other similar places)

Reply:
For the `Clear` since the writer buffer is on the heap. We're not super consistent about it elsewhere, but it's easy enough to do in new code. Arguably the encrypted data isn't sensitive but there are a couple of places where we use well known placeholder passwords, so it's effectively as secure as plaintext.

Reply:
> If this is required shouldn't that be part of Dispose or GC?

AsnWriter isn't disposable.  And the GC will eventually cause memory to be cleared, but "eventually" isn't good enough for some of our customer('s auditor)s.

This is the pattern that we already use in the other algorithms.

> public struct AsnWriter.MemoryClearScope

AsnWriter is a public type, so that'd have to go through API Review.  I don't expect it would succeed.

We could do something like that as a non-public wrapper type, but I don't think it should be introduced into this PR (if the PR was already trying it as an experiment, sure, but let's not sidetrack it with this now that it's already up).

Reply:
Some other nice things to have:
- Resize should clear (we can already add this now)
- Pinning the internal buffer - this is dangerous since it could cause a memory leak if the GCHandle isn't freed